src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -501,7 +501,12 @@ InterpreterSelectQuery::InterpreterSelectQuery(
         /// Allow push down and other optimizations for VIEW: replace with subquery and rewrite it.
         ASTPtr view_table;
         if (view)
-            view->replaceWithSubquery(getSelectQuery(), view_table, metadata_snapshot);
+        {
+            NameToNameMap parameter_values = analyzeReceiveFunctionParamValues(query_ptr);
+            query_info.is_parameterized_view = view->isParameterizedView();
```

Comment:
seems like we do not need to execute `analyzeReceiveFunctionParamValues` in case `!view->isParameterizedView()`? and the same for `view->replaceQueryParametersIfParametrizedView`? so then just put these lines under conditional statement?

Reply:
yes, good point, removed the check from the function as its now added here.
Actually I would have liked to move this part of code in replaceWithSubquery if it is not static, now we check the condition here. But better than executing unwanted code. Could be good for improvement PR to refactor StorageView.